### DISCUSSION TITLE
On_progress option in CLI

### DISCUSSION THREAD

**Discussion by Fawn06220** [JuanBindez/pytubefix#370] [Ideas]
Hi all !

Would it be possible to add the progress bar as a CLI option ?

To have something like :
`pytubefix -pb`

i'm working on a GUI (almost complete now) based on Pytubefix and i would like to capture the progress bar stdout for no console build.

Help on my project would be appreciated too, i probably made many mistakes in my code (could be optimized).

[Youtube-zik DDL](https://github.com/Fawn06220/Youtube-Zik)

**Comment by fedetony** [comment]
For my GUI I overwrite the on_progress class, see: [yt_download_pytube fix](https://github.com/fedetony/yt_download_pytubefix/blob/master/class_pytubefix_use.py)
 the class is after line 644. I use QT and emit signals, I'm not sure what you are using but you can send the progress value to whatever you want.
Enjoy :)
